src/libraries/System.Private.CoreLib/src/System/Numerics/TotalOrderIeee754Comparer.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Represents a comparison operation that compares floating-point numbers
+    /// with IEEE 754 totalOrder semantic.
+    /// </summary>
+    /// <typeparam name="T">The type of the numbers to be compared, must be an IEEE 754 floating-point type.</typeparam>
+    public struct TotalOrderIeee754Comparer<T> : IComparer<T>, IEqualityComparer<T>
+        where T : IFloatingPointIeee754<T>?
+    {
+        /// <summary>
+        /// Compares two numbers with IEEE 754 totalOrder semantic and returns
+        /// a value indicating whether one is less than, equal to, or greater than the other.
+        /// </summary>
+        /// <param name="x">The first number to compare.</param>
+        /// <param name="y">The second number to compare.</param>
+        /// <returns>
+        /// A signed integer that indicates the relative
+        /// values of <paramref name="x"/> and <paramref name="y"/>, as shown in the following table.
+        /// <list type="table">
+        ///   <listheader>
+        ///     <term> Value</term>
+        ///     <description> Meaning</description>
+        ///   </listheader>
+        ///   <item>
+        ///     <term> Less than zero</term>
+        ///     <description><paramref name = "x" /> is less than <paramref name="y" /></description>
+        ///   </item>
+        ///   <item>
+        ///     <term> Zero</term>
+        ///     <description><paramref name = "x" /> equals <paramref name="y" /></description>
+        ///   </item>
+        ///   <item>
+        ///     <term> Greater than zero</term>
+        ///     <description><paramref name = "x" /> is greater than <paramref name="y" /></description>
+        ///   </item>
+        /// </list>
+        /// </returns>
+        /// <remarks>
+        /// IEEE 754 specification defines totalOrder as a &lt;= semantic.
+        /// totalOrder(x,y) is true corresponds to the result of this method &lt;= 0.
+        /// </remarks>
+        public int Compare(T? x, T? y)
```

Comment:
The function is large/complex enough that inlining may not happen without help:
```suggestion
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public int Compare(T? x, T? y)
```

Since we always compile down to 1 path and since the `BitConverter` calls are intrinsic, each path is effectively a trivial call.